packages/fx-core/src/component/driver/deploy/azure/azureStorageDeployDriver.ts

```diff
@@ -45,18 +44,22 @@ export class AzureStorageDeployDriver implements StepDriver {
   @hooks([addStartAndEndTelemetry(ACTION_NAME, TelemetryConstant.DEPLOY_COMPONENT_NAME)])
   async run(args: unknown, context: DriverContext): Promise<Result<Map<string, string>, FxError>> {
     const impl = new AzureStorageDeployDriverImpl(args, context);
-    return wrapRun(() => impl.run(), undefined, context.logProvider);
+    return (await impl.run()).result;
   }
 
+  @hooks([addStartAndEndTelemetry(ACTION_NAME, TelemetryConstant.DEPLOY_COMPONENT_NAME)])
   execute(args: unknown, ctx: DriverContext): Promise<ExecutionResult> {
-    return wrapSummary(this.run.bind(this, args, ctx), ["driver.deploy.azureStorageDeploySummary"]);
+    const impl = new AzureStorageDeployDriverImpl(args, ctx);
+    return impl.run();
   }
 }
 
 /**
  * deploy to Azure Storage
  */
-export class AzureStorageDeployDriverImpl extends AzureDeployDriver {
+export class AzureStorageDeployDriverImpl extends AzureDeployDriverImpl {
+  protected summaries: string[] = [getLocalizedString("driver.deploy.azureStorageDeploySummary")];
+  protected summaryPrepare: string[] = [];
```

Comment:
Since summaryPrepare is always about dryrun, is it better to define it in base class, and override it in subclass?